Add unit tests to cover problematic module imports or exceptions

mataton opened the pull request:
This PR addresses #2026 by adding unit tests to ensure module imports as well as specific format exceptions are covered.

Please complete the following checklist:

* [x] I have read the [contribution guidelines](https://scikit.bio/contribute.html).

* [ ] I have documented all public-facing changes in the [changelog](https://github.com/scikit-bio/scikit-bio/blob/main/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/scikit-bio/scikit-bio/blob/main/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied.

  - **It is your responsibility to disclose** code, documentation, or other content derived from external source(s). If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] This pull request does not include code, documentation, or other content derived from external source(s).

Note: [This document](https://scikit.bio/devdoc/review.html) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

qiyunzhu commented on skbio/io/tests/test_format_imports.py:
```diff
@@ -0,0 +1,77 @@
+
+        for name, exception_cls in io_exceptions.items():
+            try:
+                ex = exception_cls(f"Test raising {name}")
```
This part looks confusing to me. Maybe some comments in the code or in the PR dialog will help?

mataton commented:
@qiyunzhu I added some comments and removed the redundant message from the portion of the test you commented on. For testing exceptions like `NewickFormatError` other format errors, we are only checking that the error can be instantiated and raised. But we are checking this for all registered formats. Checking that a specific error *is* raised when expected is left to the unit tests for that specific format.
